editors/code/src/config.ts

```diff
@@ -16,45 +16,48 @@ export interface CargoFeatures {
     allFeatures: boolean;
     features: string[];
 }
-
 export class Config {
-    langServerSource!: null | BinarySource;
+    private static readonly rootSection = "rust-analyzer";
+    private static readonly requiresReloadOpts = [
+        "cargoFeatures",
+        "cargo-watch",
+    ]
+    .map(opt => `${Config.rootSection}.${opt}`);
+
+    private cfg!: vscode.WorkspaceConfiguration;
+
+    private refreshConfig() {
+        this.cfg = vscode.workspace.getConfiguration(Config.rootSection);
+        console.log("Using configuration:", this.cfg);
```

Comment:
This will give us a better view on user's config in bug reports:
![image](https://user-images.githubusercontent.com/36276403/74477450-013bd680-4eb4-11ea-8136-c21345d352fd.png)

Reply:
I'm not a fan of it. It's useful, but it gets pretty annoying if every extension does that. I mentioned it before, we could have a "report issue" command that dumps that stuff among other things.

Reply:
Hmm, I wish we setup some logging-to-file infrastructure. There is a dedicated [`ExtensionContext.logPath`](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext) property for this. But I think @matklad would like to keep console logging at max, though not sure...

Reply:
Let's do this (in a separate PR):

* define a `function log` somewhere, which just forwards to `console.log`
* use log everywhere
* add some config to opt-in into logging

I do think this is useful to print lots of stuff. But printing should be abstracted away, so that it's easier to redirect, silence, elevate to gui popups, etc. Ideally, we should use a **built in** logger, but there isn't one afaik, so we need to do our own.